packages/entity/src/EntityQueryContextProvider.ts

```diff
@@ -23,24 +24,26 @@ export default abstract class EntityQueryContextProvider {
   /**
    * Vend a transaction runner for use in runInTransactionAsync.
    */
-  protected abstract createTransactionRunner<T>(): (
-    transactionScope: (queryInterface: any) => Promise<T>
-  ) => Promise<T>;
+  protected abstract createTransactionRunner<T>(
+    transactionConfig?: TransactionConfig
+  ): (transactionScope: (queryInterface: any) => Promise<T>) => Promise<T>;
 
   protected abstract createNestedTransactionRunner<T>(
-    outerQueryInterface: any
+    outerQueryInterface: any,
+    transactionConfig?: TransactionConfig
   ): (transactionScope: (queryInterface: any) => Promise<T>) => Promise<T>;
 
   /**
    * Start a transaction and execute the provided transaction-scoped closure within the transaction.
    * @param transactionScope - async callback to execute within the transaction
    */
   async runInTransactionAsync<T>(
-    transactionScope: (queryContext: EntityTransactionalQueryContext) => Promise<T>
+    transactionScope: (queryContext: EntityTransactionalQueryContext) => Promise<T>,
+    transactionConfig?: TransactionConfig
```

Comment:
suggestion to make this default to an empty object so that the rest of the methods can take a non-nullable type:
```suggestion
    transactionConfig: TransactionConfig = {}
```

Reply:
Because we need it to be nullable for `EntityTransactionalQueryContext.runInTransactionIfNotInTransactionAsync` (asserting empty object isn't as clear), it turns out it is cleaner to keep this optional. (tried the non-optional stuff though)